### ISSUE TITLE
SPM8 incompatible scan selection dialogue

### ISSUE THREAD

**Issue by jacob-lee** [NISOx-BDI/SwE-toolbox#99]
Matlab 2015.a
SPM8
SwE (release 2.1.0)
-------------------
When selecting scans using the file dialogue in SwE (release 2.1.0), I get the following stack trace:

```
Operands to the || and && operators must be convertible to logical scalar values.
Error in cfg_files/subsasgn_check (line 41)
                    if strcmpi(item.filter,'image') || strcmpi(item.filter,'nifti')
Error in cfg_files/subsasgn (line 80)
            [ok val] = subsasgn_check(item,subs,val1);
Error in cfg_item/setval (line 55)
        item = subsasgn(item, substruct('.','val'), {val});
Error in cfg_util (line 876)
            cm = setval(cm, varargin{4}, false);
Error in cfg_ui>local_setvaledit (line 927)
    cfg_util('setval', udmodlist.cjob, udmodlist.id{cmod}, udmodule.id{citem}, val);
Error in cfg_ui>local_valedit_SelectFiles (line 956)
    local_setvaledit(hObject, val);
Error in cfg_ui>module_Callback (line 1416)
            local_valedit_SelectFiles(hObject);
Error in gui_mainfcn (line 95)
        feval(varargin{:});
Error in cfg_ui (line 53)
    gui_mainfcn(gui_State, varargin{:}); 
Error while evaluating UIControl Callback
```

With older versions of SwE, I got no such error.

Replacing spm8 with spm12 eliminates the error.

There is a difference in SPM's cfg_files/subasgn_check.m between spm8 and spm12 that accounts for the difference: namely that `strcmpi(item.filter,'image') || strcmpi(item.filter,'nifti')` is not executed is spm12. The proximate cause of the failure is that item.filter is not a scalar, e.g. {'image', 'mat'}, so that the result of the strcmp is a 1x2 vector.

**Comment by nicholst** [comment]
@jacob-lee: Thanks so much for the detailed and insightful bug report.  While we do our best to maintain backwards compatibility with older versions of we don't have resources to do testing for anything but the current SPM12.

Anyway, I'm a bit mystified about the error... the code to select either 'image' or 'mat' has been around since last April -- see e.g. https://github.com/NISOx-BDI/SwE-toolbox/blob/v1.2.8/swe_cfg_design.m#L29 ... I'm not sure what happened with the latest change.

@TomMaullin - Do you have any ideas?  The bug relates to the matlabbatch configuration setup, now in swe_config_*model.m.

**Comment by jacob-lee** [comment]
Sure.

btw, http://www.nisox.org/Software/SwE/ indicates that SwE is a toolbox for SPM 8 and 12. May want to amend that if no longer testing for compatibility with 8.

Best,

**Comment by nicholst** [comment]
Fair point!  We'll update that.

Can you offer a datapoint:  Do many studies in your lab still use SPM8?

**Comment by jacob-lee** [comment]
Most entirely new analyses are in SPM12; however we have multi-year projects and I've kept using 8 on those for the sake of consistency (and to avoid any possible compatibility issues).

**Comment by TomMaullin** [comment]
Hi @nicholst , @jacob-lee ,

Apologies, for the delayed response, @jacob-lee can I confirm you are suggesting the root of the issue is from [this line of the configuration](https://github.com/NISOx-BDI/SwE-toolbox/blob/master/swe_config_smodel.m#L31). If I understand correctly is the issue that whilst in SPM 12 the below works:

`scans.filter = {'image', 'mat'};`

In SPM 8 the two below lines would work but the above would not:

`scans.filter = 'image';` 
`scans.filter = 'mat';`

Can I ask, do you know if the following would give the same issue:

```
scans.filter  = 'any';
scans.ufilter = '.*(image|mat)';
```

You may be able to test these on your setup by replacing [lines 31-32](https://github.com/NISOx-BDI/SwE-toolbox/blob/master/swe_config_smodel.m#L31-L32) of `swe_config_smodel.m` with the above two lines. Does this resolve the issues you have experienced in any way?

**Comment by jacob-lee** [comment]
Your suggested changes to swe_config_smodel.m appear to do the trick using both SPM8 and SPM12. Bonus is that .hdr files are no longer visible in the file dialogue.

**Comment by TomMaullin** [comment]
Ah great! I will make a pull request to update the SwE code to include this fix and ensure, after I've done a few precautionary checks, that it is in the next release.

In the meantime for your purposes I'd suggest using the fix above; I will add a comment on here once the next release is out and/or if any further related issues arise!

**Comment by nicholst** [comment]
This is addressed with PR #100
